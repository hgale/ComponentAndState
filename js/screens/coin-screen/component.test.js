import * as React from 'react'
import 'react-native'
import * as renderer from 'react-test-renderer'


import Coins from './component'

it('renders correctly', () => {
    const coinData = [
        {
            "id": "bitcoin",
            "name": "Bitcoin",
            "symbol": "BTC",
            "rank": "1",
            "price_usd": "10153.7",
            "price_btc": "1.0",
            "24h_volume_usd": "8078040000.0",
            "market_cap_usd": "170964192962",
            "available_supply": "16837625.0",
            "total_supply": "16837625.0",
            "max_supply": "21000000.0",
            "percent_change_1h": "0.33",
            "percent_change_24h": "-0.96",
            "percent_change_7d": "-10.74",
            "last_updated": "1517440467"
        },]    
  const tree = renderer.create(<Coins coins={coinData} />)
  expect(tree).toMatchSnapshot()
})