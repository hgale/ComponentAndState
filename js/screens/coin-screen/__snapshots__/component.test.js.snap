// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "marginBottom": 20,
        "marginTop": 50,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
       COIN PRICES 
    </Text>
  </View>
  <RCTScrollView
    data={
      Array [
        Object {
          "24h_volume_usd": "8078040000.0",
          "available_supply": "16837625.0",
          "id": "bitcoin",
          "last_updated": "1517440467",
          "market_cap_usd": "170964192962",
          "max_supply": "21000000.0",
          "name": "Bitcoin",
          "percent_change_1h": "0.33",
          "percent_change_24h": "-0.96",
          "percent_change_7d": "-10.74",
          "price_btc": "1.0",
          "price_usd": "10153.7",
          "rank": "1",
          "symbol": "BTC",
          "total_supply": "16837625.0",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={undefined}
        >
          <View
            style={
              Object {
                "display": "flex",
                "flexDirection": "row",
              }
            }
          >
            <Image
              source={
                Object {
                  "uri": "https://res.cloudinary.com/da7jhtpgh/image/upload/v1508609483/bitcoin_eqld4v.png",
                }
              }
              style={
                Object {
                  "height": 35,
                  "width": 35,
                }
              }
            />
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontWeight": "bold",
                  "marginLeft": 20,
                  "marginRight": 5,
                  "marginTop": 10,
                }
              }
            >
              BTC
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "marginTop": 10,
                }
              }
            >
              |
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "marginLeft": 5,
                  "marginRight": 20,
                  "marginTop": 10,
                }
              }
            >
              Bitcoin
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontWeight": "bold",
                  "marginLeft": "auto",
                  "marginRight": 10,
                  "marginTop": 10,
                }
              }
            >
              10153.7
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontWeight": "bold",
                    "marginRight": 5,
                  }
                }
              >
                 $ 
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
              >
                24h:
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "#DD2C00",
                      "fontWeight": "bold",
                      "marginLeft": 5,
                    }
                  }
                >
                   
                  -0.96
                   %
                   
                </Text>
              </Text>
            </Text>
          </View>
          <View
            style={
              Object {
                "borderBottomColor": "black",
                "borderBottomWidth": 1,
                "marginBottom": 15,
                "marginTop": 15,
              }
            }
          />
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
